### ISSUE TITLE
sql: can't insert self reference

### ISSUE THREAD

**Issue by BramGruneir** [cockroachdb/cockroach#32610]
I'm kind of surprised there isn't an issue for this yet.

```sql
CREATE TABLE bah (
  id INT NOT NULL,
  id2 INT NULL,
  CONSTRAINT "primary" PRIMARY KEY (id ASC),
  CONSTRAINT aaa FOREIGN KEY (id2) REFERENCES bah (id),
  INDEX bah_auto_index_aaa (id2 ASC),
  FAMILY "primary" (id, id2)
)

INSERT INTO bah VALUES (1,1);
pq: foreign key violation: value [1] not found in bah@primary [id]
```

This fails because we eagerly check the FKs instead of waiting into the end of the statement.  But even if we did, I think we would still run into #20041.
That being said, it might be possible to elide this FK check if the values match in the mutation path.

**Comment by knz** [comment]
Can you explain how this issue is not a duplicate of #20041? they seem to exercise the same code path and produce the same error -- I don't see the difference.

**Comment by BramGruneir** [comment]
As suggested, I'm going to merge the two into one.  But this is a slightly different mechanism.

**Comment by BramGruneir** [comment]
To be clear, this is not yet fixed, just merged this issue with #20041.
